### ISSUE TITLE
docs restructure — move the design document to `docs/design.md`, rewrite `README.md` as a user-facing page, and keep all 37 in-repo README citations resolving

### ISSUE THREAD

**Issue by Ryo-wtnb11** [Ryo-wtnb11/TeNeT-py#116]
**Goal**

`README.md` is 3046 lines and 57 top-level sections. It is the design document — invariants 1–12, milestones M1–M11, "Structure-changing differentiation", "Why retain the coupled-sector matrix representation?", "quimb in practice", the proposed package structure — and it is also, per `pyproject.toml`'s `readme = "README.md"`, the PyPI long description and the first thing anyone who lands on the repo reads. Those are two different documents wearing one filename. Someone arriving to *use* the library has to scroll past the categorical justification for keeping `side` and `dual` independent before finding out how to install it, and there is no install section, no quickstart, and no examples table to find.

User direction, verbatim:

> README設計は、doc上のどこかに移動して、READMEはもっとユーザー向けの話にしたほうが良い

So: **move the design document to `docs/design.md` unedited, and write a new user-facing `README.md`.** The reference points for both halves are `symmray` (the closest cousin — same `autoray` multi-backend block-sparse premise, README is pitch + features + example + links, install deferred to docs), `quimb` (badges + one-paragraph pitch + docs link, zero inline code), `yastn` (short, docs-first, ends with install/tests/citation) and `tenpy` (tagline + features + install + docs + citation + license).

The hard constraint is not the writing. It is that **37 lines in the repo cite the README by name and by section**, and every one of them must still resolve afterwards:

```
src/                 20   e.g. src/tenet/symmetry/base.py:206 'README "Structure-changing
tests/                8        differentiation", invariants 9 and 10'; ops/linalg.py:757
examples/             1        'README Milestone 7's'; network/__init__.py:26 'README
AGENTS.md             4        invariant 9'; tests/ops/test_contraction_plan.py:7
REPOSITORY_RULES.md   4        'README's "Plan caching" section promises'
```

Two facts make this cheap, and both were checked:

- **No test reads `README.md` as a file.** `grep -rn "README.md" tests/` is empty; the only path-walking tests are `tests/test_tensor_properties.py` (walks `src/tenet`) and `tests/network/test_hygiene.py` (walks `src/tenet/network`). So the citations are prose only, and a prose citation keeps resolving as long as the *section names* survive.
- **The README contains zero internal anchor links** (`grep -c '](#'` → 0) and zero relative links into `src/`/`tests/`/`examples/`. There is nothing to rewrite on the way out except the file's own path.

Hence the shape of the work: `git mv README.md docs/design.md` with **no content edits**, one mechanical `README` → `docs/design.md` pass over those 37 citation lines, one amendment to `REPOSITORY_RULES.md`'s "Source of truth" rule, and one new README written from scratch.

**Design sketch**

```
README.md              NEW, ~120 lines, user-facing (see skeleton below)
docs/
    design.md          git mv README.md docs/design.md  -- byte-identical body
    index.md           deferred to docs-b, see "Milestone plan"
REPOSITORY_RULES.md    "Source of truth" amended
AGENTS.md              4 lines repointed
tests/test_readme.py   NEW, ~20 lines: extracts and execs the quickstart block
```

*New `README.md` skeleton*, taken from the references with the source of each part named:

1. **Title + one-sentence tagline** — tenpy's shape. The existing line 3 already is one: "Non-Abelian symmetric tensors with ndarray-style Python APIs and backend-native numerical execution."
2. **Pitch paragraph, ~4 sentences** — symmray's register, which is the one to copy almost literally in structure: *"`symmray` provides block-sparse arrays with abelian symmetries and fermionic signs. Its arrays follow the NumPy interface where possible and can store blocks using NumPy, PyTorch, JAX, or another `autoray` backend."* Tenet's differentiator is the word symmray's sentence excludes: **non-Abelian**. Then one line linking `docs/design.md` for the why.
3. **Status** — the existing `> **Status:** early design and implementation. The API is not stable.` moves up here verbatim. Kept prominent, per `REPOSITORY_RULES.md`'s "API is unstable pre-1.0".
4. **Install** — yastn/tenpy have it, quimb and symmray defer it to docs; tenet has no docs site in docs-a, so it stays in the README: `uv add tenet-py` / `pip install tenet-py`, plus the two real extras `[jax]` and `[torch]` from `pyproject.toml`, each with the one-line reason already in the file (jax≥0.10 for the wide-matrix qr JVP; torch is eager-only).
5. **Quickstart** — symmray's README has one substantial block (U(1) indices → contraction → reshape → SVD) and it is the single most useful thing in that README. Tenet's should mirror it and end where tenet goes further, at `jit(grad(...))`. Sketch — **the merged text must be the exact string the test execs**, and the spellings come from `tests/ops/test_einsum.py` and `examples/dmrg.py`, not from this issue:

   ```python
   import tenet
   from tenet import IN, OUT, Leg, SymmetricTensor

   # an SU(2) leg, a random invariant tensor, an ndarray-style contraction
   a = SymmetricTensor.random((Leg(v, OUT), Leg(w, OUT), Leg(u, OUT)), seed=0)
   b = SymmetricTensor.random((Leg(u, OUT, dual=True), Leg(w, IN), Leg(v, IN)), seed=1)
   c = tenet.einsum("abc,cde->abde", a, b)     # legs, not codomain x domain
   q, s, r = tenet.linalg.svd(c, axes=((0, 2), (1, 3)))

   # the same tensors are a JAX pytree; structure is static, blocks are traced
   import tenet.pytree
   g = jax.jit(jax.grad(lambda t: tenet.norm(t) ** 2))(a)
   ```
6. **Features** — the honest list, each one a thing that exists today: exact SU(2) recoupling coefficients with vendored gauge fixtures; `to_dense()` as the traceable dense boundary and no implicit densification anywhere else; `tenet.ad` (custom SVD/eigh rules); the `svd_truncated`-outside / `svd(bond=)`-inside pairing that lets truncation live outside a trace and fixed-structure factorization inside one; `tenet.network` (`MPS`/`MPO`/`dmrg`, `CTMEnv`/`converge`/`unrolled`); NumPy/JAX/PyTorch through `autoray`, torch eager-only.
7. **Examples table** — three rows, each linking the file and naming its oracle, because the oracle is the claim: `examples/dmrg.py` (U(1) Heisenberg vs. exact diagonalization), `examples/ctmrg.py` (classical Ising vs. Onsager, then a U(1)/SU(2) iPEPS gradient), `examples/vmc_mps.py` (symmetric MPS → pytree → grad → SGD).
8. **Related projects** — symmray's README does *not* do this (it names only its dependencies) and neither does quimb's; this is the one place to depart from the closest cousin, because tenet's position is only legible relative to its neighbours. One respectful line each: `symmray` (Abelian + fermionic, same autoray premise), `yastn` (Abelian, differentiable, sphinx-documented, a direct influence on the CTMRG example's `sym='Z2'` choice), `TensorKit.jl`, `froSTspin`. No comparison table, no scoring.
9. **Docs / design** — link `docs/design.md` and `REPOSITORY_RULES.md`. Becomes a docs-site link in docs-b.
10. **License + citation** — yastn and tenpy both end here. **There is no `LICENSE` file in this repository** (`ls LICENSE*` → no matches) and no `CITATION.cff`. This issue does not pick one. See "Key decisions".

*`REPOSITORY_RULES.md` amendment*, replacing the current bullet at lines 8–10:

```markdown
- `docs/design.md` is the design document. Code follows it; if implementation
  experience shows the design is wrong, update `docs/design.md` in the same PR
  and say why.
- `README.md` is the user-facing page, not a design document. It may fall out
  of date with the design only in the sense of omitting things, never of
  contradicting them.
```

and lines 3–4's "Design rules live in `README.md` (authoritative)" → `docs/design.md`.

**Key decisions & rationale**

- **`docs/design.md` as one file, not split into chapters.** The tempting split is `design/overview.md` + `design/invariants.md` + `design/milestones.md`, and it is wrong for this PR. 20 docstrings in `src/` cite sections by *name* ("README 'Repartitioning is different from viewing'", "README invariant 8"); with one file, `grep -n "^# Repartitioning" docs/design.md` resolves any of them in one command and the mechanical rename is a single sed. With a split, every citation needs a human decision about which chapter it landed in, inside the same PR that is supposed to be a `git mv`. The file is 3046 lines of GitHub-rendered markdown with a working sidebar TOC; that is not yet a navigation problem. Split it when the docs site exists and the nav actually hurts (docs-b or later), not while it is still a move.
- **Move, do not edit.** The acceptance criterion is mechanical: `git show HEAD:README.md | diff - docs/design.md` is empty except for the milestone-status/self-reference lines that name the file itself. Any prose improvement to the design doc is a separate PR — a 3000-line diff that is *mostly* a move is unreviewable, and this repo's rule that design changes are argued in the PR message makes smuggled edits actively costly.
- **Section headings survive verbatim, and that is a checked criterion, not an intention.** The check is one command, and it is worth writing into the PR description: `comm -13 <(git show HEAD:README.md | grep '^#') <(grep '^#' docs/design.md)` must be empty. That single line is what makes all 37 citations safe.
- **The 37 citations get renamed in the same PR.** Leaving the word "README" in `src/tenet/symmetry/base.py:206` after the move would point a reader at a file that no longer says that. It is one `sed` over 37 lines and it is in scope precisely because it is mechanical; nothing but the file name changes, since the section names do not.
- **`pyproject.toml`'s `readme = "README.md"` stays.** The PyPI long description becomes the user-facing page, which is what it should always have been. Nothing else in `pyproject.toml` changes in docs-a.
- **No docs tooling in docs-a.** GitHub renders `docs/design.md` fine — the file has no HTML, no mermaid, no admonitions, no anchor links, and its only non-vanilla construct is 29 ```` ```math ```` fences, which are GitHub's own dialect. Adding a site generator in the same PR means adding a dependency group, a config file, a CI step and a rendering regression surface to a PR whose value is the move. Ship the move; the site is docs-b.
- **When the site does happen, `mkdocs-material`, not sphinx — but with one honest caveat.** The naive argument ("sphinx wants rst") is false and worth retiring here: **symmray's docs are sphinx with 14 `.md` files** (`docs/conf.py` + `abelian_arrays.md`, `fermionic_arrays.md`, …, via MyST), and yastn and tenpy are both sphinx+RTD. So sphinx is the field's default and it does eat markdown. The caveat that actually decides it is the 29 ```` ```math ```` fences: *neither* toolchain renders GitHub's math fence out of the box. MkDocs closes it in config — `pymdownx.superfences` supports a custom fence routing `math` to `pymdownx.arithmatex`, so the fences stay byte-identical in the moved file, which is exactly the property this issue is protecting. MyST needs `dollarmath` plus either a fence rewrite or a transform. Against that, sphinx's advantage is `autodoc` maturity for the API reference; `mkdocstrings[python]` is the mkdocs answer and is adequate for a library whose docstrings are prose-heavy and already reference-styled. Recommendation: `mkdocs-material` + `pymdownx.arithmatex` + `mkdocstrings[python]`, in a `docs` dependency-group, never in core — `REPOSITORY_RULES.md`'s dependency rule is about the *runtime* dependency set and a docs group does not touch it. Revisit if the API reference turns out to need autodoc's cross-referencing.
- **CI in docs-b, as a build check before a deploy.** `mkdocs build --strict` appended as a step to the existing `test` job, following the precedent already in `.github/workflows/ci.yml` where the torch run is an appended step rather than a matrix axis ("the suite is ~60s… running everything twice buys nothing"). Pages deploy is a separate job with `permissions: pages: write`, and it is fine for it to land a PR later than the build check. A broken docs build should redden CI; a broken *deploy* should not block a merge.
- **The quickstart gets a test, and it is the only new test.** `tests/test_readme.py` reads `README.md`, extracts the fenced block(s) tagged ```` ```python ```` under the Quickstart heading, and `exec`s them in a fresh namespace. ~20 lines, no framework, no fixtures. The reason is specific rather than general: this README's whole job is to be the first thing a user runs, the API is explicitly unstable pre-1.0, and a rotted quickstart in the PyPI long description is the most expensive stale line in the repo. Notably neither quimb nor symmray tests their README block — quimb has no inline code at all and symmray's is illustrative — so this is tenet choosing stricter than the references, deliberately. It also means the block must be genuinely runnable on the core install: **no `[jax]` requirement in the executed portion**, or the JAX lines get split into a second block that the test skips when JAX is absent (mirroring how the backend tests already skip).
- **English only.** The design doc is English, the docstrings are English, the references are English. No Japanese docs in scope.
- **The license is flagged, not chosen.** There is no `LICENSE` file. Every reference project states one (tenpy: Apache-2.0 since 1.0.4; yastn ends on citation + license). A README with a "License" heading and nothing under it is worse than no heading, and picking a license is a decision with consequences the repository owner makes, not this PR. The PR should include the section with a `TODO(license)` placeholder or omit it, and the issue should be closed with an explicit note that the choice is outstanding. Same for `CITATION.cff` — tenpy and yastn both have citation sections and tenet has no paper yet.

**Milestone plan**

- **docs-a (this issue)** — `git mv README.md docs/design.md`; new user-facing `README.md`; `tests/test_readme.py`; the 37 citation renames; `REPOSITORY_RULES.md` and `AGENTS.md` amendments. No new dependencies, no CI change, no site.
- **docs-b (follow-up)** — `mkdocs-material` + `pymdownx.arithmatex` + `mkdocstrings[python]` in a `docs` dependency-group; `mkdocs.yml`; `docs/index.md` (the README, or an include of it); `mkdocs build --strict` in CI; Pages deploy job; `docs/tutorials/{dmrg,ctmrg,vmc}.md` derived from the three example module docstrings — **linking** the example files, not copying their code, since `tests/integration/` already executes those files and a copy would be an untested fork of tested code; `docs/api/` from docstrings.

Splitting this way because docs-a is reviewable by inspection (is the moved file byte-identical? do the headings match? does the quickstart run?) and docs-b is reviewable by looking at a rendered site. Merged into one PR, the reviewer has to do both at once through a diff that is 3000 lines of noise.

**References**

- `README.md` (3046 lines, 57 `^# ` sections, 29 ```` ```math ```` fences, 0 internal anchors, 0 relative repo links) → becomes `docs/design.md`
- `REPOSITORY_RULES.md:3-4,8-10,21` — the "Source of truth" rule and the commit-message rule that names README milestones
- `AGENTS.md:8,35,39,83` — "**`README.md` is the design document.**", the package-structure pointer, the condensed invariants header, the milestone-order rule
- `pyproject.toml:5` — `readme = "README.md"`
- `.github/workflows/ci.yml` — the appended-step precedent for docs-b's build check
- `examples/{dmrg,ctmrg,vmc_mps}.py` — module docstrings are the tutorial source material and already name their oracles
- symmray — README skeleton (badges, pitch, features, one substantial code block, docs links, no install, no comparison section); docs are **sphinx + MyST with `.md` sources** (`docs/conf.py`, 14 markdown pages)
- quimb — badges + one-paragraph pitch + RTD links, zero inline code, no install section; sphinx
- yastn — short and docs-first; ends with install, tests, profiling, citation; sphinx (`docs/source/`, Makefile)
- tenpy — tagline, feature list, conda install, RTD link, BibTeX citation, Apache-2.0 statement; sphinx + RTD

**Out of scope**

- Any content rewrite of the design document. Mechanical link/path fixes only.
- Splitting `docs/design.md` into chapters.
- Versioned docs, readthedocs account setup, logo, branding.
- Choosing a license or writing a `CITATION.cff` — flagged above as an owner decision.
- The docs site itself (mkdocs config, Pages deploy, API reference, tutorials) — docs-b.

**Acceptance criteria**

- `docs/design.md` exists and its body is identical to `README.md` at the parent commit, apart from lines that name the file itself.
- `comm -13 <(git show HEAD:README.md | grep '^#') <(grep '^#' docs/design.md)` is empty — every heading of the old README, at every level, exists verbatim in the moved file.
- `grep -rn "README" src/ tests/ examples/ AGENTS.md REPOSITORY_RULES.md | grep -iv "readme.md:"` returns only references that are genuinely about the new user-facing README; all 37 design-doc citations name `docs/design.md`, with their quoted section names unchanged.
- `REPOSITORY_RULES.md` "Source of truth" names `docs/design.md` as the design document and states what `README.md` is instead.
- `AGENTS.md`'s four references point at `docs/design.md`.
- The new `README.md` contains, in order: tagline, pitch, status, install (with `[jax]` and `[torch]`), quickstart, features, examples table linking all three example files, related projects, a link to `docs/design.md`.
- `tests/test_readme.py` extracts the README quickstart block and executes it; it fails if the block is edited into something that does not run.
- `pyproject.toml` still has `readme = "README.md"`; no new runtime or dev dependency is added.
- `uv run ruff check .`, `uv run ruff format --check .`, `uv run pytest` all pass; the suite is 2927 tests before this change and 2928+ after (the README quickstart test).

**Depends on**

Nothing. #114 (M11b) is merged, `main` is green at 2927 tests, and every file this touches is currently stable.
